xbmc/platform/Platform.cpp

```diff
@@ -47,3 +53,22 @@ void CPlatform::Init()
   // nothing for now
 }
 
+std::string CPlatform::GetUniqueHardwareIdentifier()
+{
+  static std::string uuid = NoValidUUID;
+
+#if defined(HAS_LINUX_NETWORK) || defined(HAS_WIN32_NETWORK)
```

Comment:
but the check is for LINUX_NETWORK ... it is true for other platforms too ... its not a check for TARGET_LINUX and TARGET_WIN32